addons/project/static/src/components/project_many2one_field/project_many2one_field.js

```diff
@@ -10,7 +10,10 @@ export class ProjectMany2OneField extends Many2OneField {
         const props = super.Many2XAutocompleteProps;
         const { project_id, parent_id } = this.props.record.data;
         if (!project_id && !parent_id) {
-            props.placeholder = _t("Private");
+            const project_id_config = this.env.model.config.activeFields.project_id;
+            if (project_id_config && project_id_config.required !== "1") {
+                props.placeholder = _t("Private");
+            }
```

Comment:
Use camelCase in js :slightly_smiling_face: 
```suggestion
            const projectIdConfig = this.env.model.config.activeFields.project_id;
            if (projectIdConfig && projectIdConfig.required !== "1") {
                props.placeholder = _t("Private");
            }
```
Also, if no project, is it false or undefined? If undefined:
```suggestion
            const projectIdConfig = this.env.model.config.activeFields.project_id;
            if (projectIdConfig?.required !== "1") {
                props.placeholder = _t("Private");
            }
```